internal/action/actions.go

```diff
@@ -160,13 +160,19 @@ func (h *BufPane) Center() bool {
 	return true
 }
 
-// MoveCursorToView moves the cursor to the current view
-func (h *BufPane) MoveCursorToView() bool {
+// CursorToView moves the cursor to the current view
+func (h *BufPane) CursorToView() bool {
 	v := h.GetView()
-	h.Cursor.GotoLoc(buffer.Loc{0, v.StartLine.Line})
-	h.RemoveAllMultiCursors()
 	h.Cursor.Deselect(true)
-	h.Relocate()
+	loc := h.Cursor.Loc
+	if(loc.Y < v.StartLine.Line || loc.Y > v.StartLine.Line + h.BufView().Height - 1) {
```

Comment:
This is, again, not the right way to check if the cursor is within the view. For the same reason: the cursor may be in the first line in the view, yet outside the view, since this line may be wrapped, so only a part of this line may be visible. That's why `v.StartLine` is a `SLoc` structure, with 2 fields: `Line` which is the number of the logical line in the file, and `Row` which is the number of the visual line within this logical line.

The right way is:

```
	sloc := h.SLocFromLoc(h.Cursor.Loc)
	if sloc.LessThan(v.StartLine) || sloc.GreaterThan(h.Scroll(v.StartLine, h.BufView().Height-1)) {
```

Reply:
Thanks, that makes sense.